### PR TITLE
added theming capability

### PR DESCRIPTION
**Pull request by lbialy**

### CODE REVIEW

File: modules/core/src/main/scala/Theme.scala [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      def nothingSelected: String      = s.underline.darkGray
+      def selected: String             = s.lightCyan
+      def selectedMany: String         = s.bold.lightCyan
+      def selectedManyInactive: String = s.underline.lightGray
+      def canceled: String             = s.red
+      def error: String                = s.bold.red
+    end extension
+  end Monokai
+
+  def Darcula: Theme = new Theme:
```
Comment (keynmol) [inline modules/core/src/main/scala/Theme.scala:108]: I think these can just be static objects, `case object Darcula extends Theme:`

Comment (keynmol) [reply modules/core/src/main/scala/Theme.scala:108]: Actually I tested a few of the themes, and to my eye they have nothing to do with respective known themes.

As such, they should be removed or re-done from scratch. The Default and NoColors can remain.

File: modules/core/src/main/scala/TextFormatting.scala [hunk 2]
```diff
@@ -16,19 +16,53 @@
 
 package cue4s
 
-class TextFormatting(enabled: Boolean):
-  private def colored(msg: String)(f: String => fansi.Str) =
-    if enabled then f(msg).toString else msg
+object TextFormatting:
+  private def colored(msg: String)(f: String => fansi.Str) = f(msg).toString
```
Comment (antonsviridov-src) [inline modules/core/src/main/scala/TextFormatting.scala:20]: Given it no longer depends on enabled flag, this should be inlined at callsites and removes – it doesn't seem to serve any purpose anymore

**Review by keynmol — APPROVED** [review]
thanks!

**Pull request merged**